serrano/resources/preview.py

```diff
@@ -100,31 +105,34 @@ def get(self, request):
             iterable = processor.get_iterable(queryset=queryset,
                                               request=request)
 
-            exported = exporter.read(iterable,
-                                     request=request,
-                                     offset=offset,
-                                     limit=limit)
+            rows = exporter.manual_read(iterable,
+                                        request=request,
+                                        offset=offset,
+                                        limit=limit)
         else:
             iterable = processor.get_iterable(queryset=queryset,
                                               request=request,
                                               limit=limit,
                                               offset=offset)
 
-            exported = exporter.read(iterable, request=request)
+            # Select the requested reader
+            reader = params['reader']
 
-        for row in exported:
-            pk = None
-            values = []
+            if reader == 'threaded':
+                method = exporter.threaded_read
+            elif reader == 'cached':
+                method = exporter.cached_read
+            elif reader == 'cached_threaded':
+                method = exporter.cached_threaded_read
+            else:
+                method = exporter.read
```

Comment:
This logic block appears a couple different places. Would it be better to add a method to exporter or a dictionary mapping the reader strings to reader methods? So something like

```python
# in exporter.py or local utility method
READER_METHODS = {
    'threaded': exporter.threaded_read,
    'cached': exporter.cached_read,
    'cached_threaded': exporter.cached_threaded_read,
}

def get_reader_method(reader):
    return READER_METHODS.get(reader, default=exporter.read)
```